packages/augur-core/source/contracts/reporting/RepPriceOracle.sol

```diff
@@ -33,92 +45,113 @@ contract RepPriceOracle is IRepPriceOracle, Initializable {
         require(cash != address(0));
         uniswapFactory = IUniswapV2Factory(_augur.lookup("UniswapV2Factory"));
         require(uniswapFactory != IUniswapV2Factory(0));
+        ensRegistry = IENSRegistry(_augur.lookup("ENSRegistry"));
+        require(ensRegistry != IENSRegistry(0));
+    }
+
+    function upgradeUniswapFactory() public {
+        require(!uniswapUpgraded, "Already upgraded");
+        IENSResolver _resolver = IENSResolver(ensRegistry.resolver(UNISWAP_REGISTRY_ENS_NAME));
+        require(_resolver != IENSResolver(0), "Resolver for ENS name not found");
+        address _newUniswapFactory = _resolver.addr(UNISWAP_REGISTRY_ENS_NAME);
+        require(_newUniswapFactory != address(0), "Could not get address from resolver");
+        uniswapFactory = IUniswapV2Factory(_newUniswapFactory);
+        uniswapUpgraded = true;
     }
 
     // TODO: Consider when this should be called other than when the price is requested as part of new fee setting
     function pokeRepPriceInAttoCash(IV2ReputationToken _reputationToken) external returns (uint256) {
-        if (exchangeData[address(_reputationToken)].exchange == IUniswapV2(0)) {
+        if (exchangeData[address(_reputationToken)].exchange == IUniswapV2Exchange(0)) {
             initializeUniverse(_reputationToken);
         }
         ExchangeData memory _newExchangeData = calculateNewExchangeData(_reputationToken);
         exchangeData[address(_reputationToken)] = _newExchangeData;
```

Comment:
I believe this might be extremely gas-inefficient. Because you are taking memory struct and assigning to storage, all 7 variables in the struct must be rewritten to storage, even if none of them require updates. Even in the case where some updates are required, it will rewrite these attributes each time: exchange, repIsToken0, upgradeHandled, even though they never or rarely change. If the calculate could be the one that actually performs the update, it could be more efficient.

Maybe not important since the cadence of calling this function is low?